### ISSUE TITLE
Movement Follow Broken [Fabric] - 1.8.6

### ISSUE THREAD

**Issue by AwesomeAceStation** [samolego/Taterzens#89]
**Describe the bug**
NPC movement for "follow" broken. After leaving line of sight using Nether Portal or after exiting save game and reentering world. Command '/npc edit movement follow PLAYERS' fixes problem temporarily. Using presets or removing NPC does not work.

**Platform(s)**

- [x] Fabric
- [ ] Forge

**Environment(s)**

- [x] Singleplayer
- [ ] Server

**Versions**
* Minecraft 1.18.2
* Taterzens 1.8.6

**To Reproduce**
1. Create NPC and set movement type
2. Set NPC movement to follow 'PLAYERS'
3. Leave NPC line of sight using Nether Portal or exiting game and reentering world

**Additional context**
N/A

**Comment by samolego** [comment]
> Command '/npc edit movement follow PLAYERS' fixes problem temporarily

>Steps to reproduce the behavior:

>1. Create NPC and set movement type
>2. Set NPC movement to follow 'PLAYERS'

Which follow type is broken then? UUID one?

**Comment by AwesomeAceStation** [comment]
> So following players works until the 3rd step, then after reentering command works again?

Yes. I also tried applying the movement follow command and save the preset before exiting the saved world as well; "Follow" gets reset in the JSON file to blank (not saving the preset, under "Follow" the Type: "PLAYERS" or UUID is still listed in JSON). I have also tried using 'follow' with UUID as well but as soon I as do step 3, NPC does not follow and behaves like the 'LOOK' command where NPC is stationary and looks at player time to time.

**Comment by James103** [comment]
What do you get in the Minecraft log when you do this? If you get any exceptions or error messages, can you please post them here?

**Comment by AwesomeAceStation** [comment]
> What do you get in the Minecraft log when you do this? If you get any exceptions or error messages, can you please post them here?
[latest.log](https://github.com/samolego/Taterzens/files/8447039/latest.log)

I don't really see any errors about the mod besides info about my texture pack and minor player location logs.
